app/Get.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Get where
+
+import Control.Monad.IO.Class
+import Data.Text (Text, pack, unpack)
+import Data.Aeson
+import Network.HTTP.Req
+import Fragnix.Slice (Slice, writeSlice, sliceNestedPath, moduleNameReference)
+import Fragnix.Core.Slice (uses, reference, Reference(OtherSlice, Builtin))
+import Fragnix.Environment (writeSymbols)
+import Fragnix.Paths (slicesPath, environmentPath)
+import Language.Haskell.Names (Symbol, symbolModule)
+import Language.Haskell.Exts (ModuleName (..), prettyPrint)
+import Data.Maybe (isJust)
+import Text.Read (readMaybe)
+import Control.Monad (forM_, filterM)
+import System.Directory (createDirectoryIfMissing, doesFileExist)
+import System.FilePath ((</>))
+import Utils (IDType (SliceID, EnvID), sliceRequest, envRequest)
+
+get :: IDType -> String -> Bool -> IO ()
+get SliceID = fetchSlice
+get EnvID = fetchEnv
+
+fetchSlice :: String -> Bool -> IO ()
+fetchSlice id nodeps = do
+  sliceExists <- doesSliceExist id
```

Comment:
The following bug is possible here:

We first call `fetchSlice 12345 False` and get slice 12345 on disk without its dependencies.
We then call `fetchSlice 12345 True`, see that slice 12345 is already on disk and fail to get its dependencies.